### PR TITLE
Auto Increment Refactor

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: memory/table.go [hunk 1]
```diff
@@ -492,20 +492,23 @@ func rowsAreEqual(ctx *sql.Context, schema sql.Schema, left, right sql.Row) (boo
 }
 
 // PeekNextAutoIncrementValue peeks at the next AUTO_INCREMENT value
-func (t *Table) PeekNextAutoIncrementValue(*sql.Context) (interface{}, error) {
```
Comment (zachmu) [inline memory/table.go:495]: Problem is this could be a float. If you go this route you need affordances for that

Comment (zachmu) [reply memory/table.go:495]: Same comment throughout

File: sql/information_schema/information_schema.go [hunk 2]
```diff
@@ -810,7 +809,7 @@ func tablesRowIter(ctx *Context, cat Catalog) (RowIter, error) {
 				nil,                        // max_data_length
 				nil,                        // max_data_length
 				nil,                        // data_free
-				autoVal,                    // auto_increment
+				nil,                        // auto_increment (always nil)
```
Comment (zachmu) [inline sql/information_schema/information_schema.go:812]: This isn't right, MySQL definitely returns accurate info for this. Why do we need to get rid of it?

**Review by zachmu — APPROVED** [review]
Mostly fine, but you need to fix the information schema thing you broke (or at least add a skipped test for it)

File: sql/information_schema/information_schema.go (hunk 2 above)
Comment (andrew-wm-arthur) [reply sql/information_schema/information_schema.go:812]: mysql always returns NULL for user space tables

**Pull request merged**